src/classes/QueryBuilder.cls

```diff
@@ -166,11 +180,12 @@ public without sharing class QueryBuilder extends NebulaCore implements IQueryBu
 
         this.addLogEntry(query);
 
-        return query;
+        return this.query;
     }
 
     private void addCommonQueryFields() {
         if(!NebulaSettings.repositorySettings.IncludeCommonFields__c) return;
+        if(!this.shouldAddCommonFields) return;
```

Comment:
the custom setting is almost certainly the better implementation. I haven't had the time to flesh out the required custom settings in my
own org but that's one of my top priorities once I'm back from my trip